components/admin_console/compliance_reports/compliance_reports.tsx

```diff
@@ -1,57 +1,70 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedDate, FormattedMessage, FormattedTime} from 'react-intl';
 import {Client4} from 'mattermost-redux/client';
+import {Compliance} from 'mattermost-redux/types/compliance';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {Dictionary} from 'mattermost-redux/types/utilities';
 
 import LoadingScreen from 'components/loading_screen';
 import ReloadIcon from 'components/widgets/icons/fa_reload_icon';
 import LocalizedInput from 'components/localized_input/localized_input';
 import {t} from 'utils/i18n.jsx';
 
-export default class ComplianceReports extends React.PureComponent {
-    static propTypes = {
+type Props = {
 
-        /*
-         * Set if compliance reports are licensed
-         */
-        isLicensed: PropTypes.bool.isRequired,
+    /*
+        * Set if compliance reports are licensed
+        */
+    isLicensed: boolean;
 
-        /*
-         * Set if compliance reports are enabled in the config
-         */
-        enabled: PropTypes.bool.isRequired,
+    /*
+        * Set if compliance reports are enabled in the config
+        */
+    enabled: boolean;
 
-        /*
-         * Array of reports to render
-         */
-        reports: PropTypes.arrayOf(PropTypes.object).isRequired,
-        users: PropTypes.object.isRequired,
+    /*
+        * Array of reports to render
+        */
+    reports: Compliance[];
+    users: Dictionary<UserProfile>;
 
-        /*
-         * Error message to display
-         */
-        serverError: PropTypes.string,
+    /*
+        * Error message to display
+        */
+    serverError?: string;
 
-        readOnly: PropTypes.bool,
+    readOnly?: boolean;
 
-        actions: PropTypes.shape({
+    actions: {
 
-            /*
-             * Function to get compliance reports
-             */
-            getComplianceReports: PropTypes.func.isRequired,
+        /*
+            * Function to get compliance reports
+            */
+        getComplianceReports: () => Promise<{data: Compliance[]}>;
 
-            /*
-             * Function to save compliance reports
-             */
-            createComplianceReport: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+        /*
+            * Function to save compliance reports
+            */
+        createComplianceReport: (job: Partial<Compliance>) => Promise<{data: Compliance, error?: Error}>;
```

Comment:
I'm not sure if this is a bug in `mattermost-webapp` or `mattermost-redux`, but the [createComplianceReport()](https://github.com/mattermost/mattermost-redux/blob/5aa938486c49d76f55953dca7890fb6f4ad601b7/src/client/client4.ts#L2656) function in `mattermost-redux` takes in a `Job`, however, if you look at the type definition of [Job](https://github.com/mattermost/mattermost-redux/blob/5aa938486c49d76f55953dca7890fb6f4ad601b7/src/types/jobs.ts#L6), it does not match the usage in [runReport()](https://github.com/mattermost/mattermost-webapp/pull/7036/files#diff-404f3aee8367f0adaedc23366b864a9ce42e47b0fea4cec46fa15421e36f403dR104-R109) below, which is why I used `Partial<Compliance>` instead.

Reply:
Looks like this is a bug in `mattermost-redux`. The [`/compliance/reports`](https://github.com/mattermost/mattermost-server/blob/v5.28.1/api4/compliance.go#L23) API in `mattermost-server` is expecting a [Compliance](https://github.com/mattermost/mattermost-redux/blob/5aa938486c49d76f55953dca7890fb6f4ad601b7/src/types/compliance.ts), not a [Job](https://github.com/mattermost/mattermost-redux/blob/5aa938486c49d76f55953dca7890fb6f4ad601b7/src/types/jobs.ts#L6), see:
- https://github.com/mattermost/mattermost-server/blob/v5.28.1/model/compliance.go#L101
- https://github.com/mattermost/mattermost-server/blob/v5.28.1/model/compliance.go#L24

Reply:
Created a separate PR for this at mattermost/mattermost-redux#1289